metadata_service/proxy/neo4j_proxy.py

```diff
@@ -853,57 +854,75 @@ def _get_user_resource_relationship_clause(relation_type: UserResourceRel, id: s
             raise NotImplementedError(f'The relation type {relation_type} is not defined!')
         return relation
 
-    @staticmethod
-    def _get_user_table_relationship_clause(relation_type: UserResourceRel, tbl_key: str = None,
-                                            user_key: str = None) -> str:
-        """
-        Returns the relationship clause of a cypher query between users and tables
-        The User node is 'usr', the table node is 'tbl', and the relationship is 'rel'
-        e.g. (usr:User)-[rel:READ]->(tbl:Table), (usr)-[rel:READ]->(tbl)
+    @timer_with_counter
+    def get_dashboard_by_user_relation(self, *, user_email: str, relation_type: UserResourceRel) \
+            -> Dict[str, List[DashboardSummary]]:
         """
-        tbl_matcher: str = ''
-        user_matcher: str = ''
+        Retrieve all follow the Dashboard per user based on the relation.
 
-        if tbl_key is not None:
-            tbl_matcher += ':Table'
-            if tbl_key != '':
-                tbl_matcher += f' {{key: "{tbl_key}"}}'
+        :param user_email: the email of the user
+        :param relation_type: the relation between the user and the resource
+        :return:
+        """
+        rel_clause: str = self._get_user_resource_relationship_clause(relation_type=relation_type,
+                                                                      id='',
+                                                                      resource_type=ResourceType.Dashboard,
+                                                                      user_key=user_email)
 
-        if user_key is not None:
-            user_matcher += ':User'
-            if user_key != '':
-                user_matcher += f' {{key: "{user_key}"}}'
+        query = textwrap.dedent(f"""
+        MATCH {rel_clause}<-[:DASHBOARD]-(dg:Dashboardgroup)<-[:DASHBOARD_GROUP]-(clstr:Cluster)
+        OPTIONAL MATCH (resource)-[:DESCRIPTION]->(dscrpt:Description)
+        OPTIONAL MATCH (resource)-[:EXECUTED]->(last_exec:Execution)
+        WHERE split(last_exec.key, '/')[5] = '_last_successful_execution'
```

Comment:
could you add a comment on this query? 
like `split(last_exec.key, '/')[5] = '_last_successful_execution'` not super clear to me.